### ISSUE TITLE
Bug - ModuleDict ordering issue

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#298]
Original test - `test/test_nn.py::TestNN::test_ModuleDict` from Pytorch

Repro

~~~
import torch
import torchdynamo
import collections
from collections import OrderedDict

@torchdynamo.disable
def check(module_dict, modules):
    assert len(module_dict) == len(modules)
    for k1, m2 in zip(modules, module_dict.children()):
        print("-->", k1, modules[k1], m2)
        assert modules[k1] is m2

def fn():
    modules = OrderedDict([
        ('act', torch.nn.ReLU()),
    ])
    module_dict = torch.nn.ModuleDict(modules)


    next_modules = {
        'fc4': torch.nn.Linear(5, 6),
        'act3': torch.nn.Sigmoid()
    }
    modules.update(next_modules.items())
    module_dict.update(next_modules)
    check(module_dict, modules)

fn()


torchdynamo.config.debug = True
torchdynamo.config.trace = True
with torchdynamo.optimize("eager"):
    fn()
~~~

@yanboliang if I recall correctly, you touched some code related to `OrderedDict`. FYI, in case you are interested

**Comment by yanboliang** [comment]
This is a bug because we don't use the original order to reconstruct dict. https://github.com/pytorch/torchdynamo/pull/308 will fix it.
